Lib/contextlib.py

```diff
@@ -469,3 +469,24 @@ def _fix_exception_context(new_exc, old_exc):
                 exc_details[1].__context__ = fixed_ctx
                 raise
         return received_exc and suppressed_exc
+
+
+class nullcontext(AbstractContextManager):
+    """Context manager that does no additional processing.
+
+    Used as a stand-in for a normal context manager, when a particular
+    block of code is only sometimes used with a normal context manager:
+
+    cm = optional_cm if condition else nullcontext()
+    with cm:
+        # Perform operation, using optional_cm if condition is True
+    """
+
+    def __init__(self, thing=None):
```

Comment:
As per the docs comment, replace `thing` with `enter_result` for both the parameter and attribute name.